src/System.Runtime.Extensions/src/System/AppDomain.cs

```diff
@@ -26,5 +27,147 @@ public partial class AppDomain
                 AppContext.UnhandledException -= value;  
             }  
         }  
+
+        public string DynamicDirectory => null;
+        public string FriendlyName => "DefaultDomain";
+        public int Id => DefaultADID;
+        public bool IsFullyTrusted => true;
+        public bool IsHomogenous => true;
+        public bool ShadowCopyFiles => false;
+        public event System.EventHandler DomainUnload
+        {
+            add  
+            {  
+                throw new PlatformNotSupportedException();
```

Comment:
You want to minic the behavior that you get in the full framework in default domain. You can subscribe to the event in default domain just fine, but your callback won't be ever called.